### PR TITLE
PD-2851 Update Req to 0.5

### PR DESCRIPTION
**Pull request by nemanjabogdanovic**
Not sure if I missed something with the Action setup.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -47,7 +49,10 @@ defmodule WorkflowEngine.MixProject do
 
   defp aliases do
     [
-      "test.including_external": ["test --include external_service"]
+      "test.including_external": ["test --include external_service"],
+      "test.ci": ["test --color --exclude external:true"],
+      "test.ci.watch": ["test.watch --color --exclude external:true"],
+      "test.dev": ["test.watch --trace"]
```
Comment (iStefo) [inline mix.exs:55]: Do we need these additional targets?

File: lib/actions/http.ex [hunk 2]
```diff
@@ -250,7 +250,12 @@ defmodule WorkflowEngine.Actions.Http do
     end)
   end
 
-  @spec unwrap_response({:error, Mint.TransportError.t()} | {:ok, Req.Response.t()}, any) ::
+  @spec unwrap_response(
+          {:error, Req.TransportError.t()}
```
Comment (iStefo) [inline lib/actions/http.ex:254]: Does this type spec work when only req 0.3 is installed? I would assume this leads to problems

File: .github_workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    steps:
+      - name: Check out Repository
+        uses: actions/checkout@v4
+
+      - name: Set up Elixir
+        uses: erlef/setup-beam@v1
+        with:
+          otp-version: ${{matrix.otp}}
+          elixir-version: ${{matrix.elixir}}
+
```
Comment (iStefo) [inline .github_workflows/ci.yml:34]: BXDK is a dependency in dev/test, but checking it out needs the deploy key present. See e.g. batch service pipeline for SSH key setup (do it before the mix install)

**Review by iStefo — CHANGES_REQUESTED** [review]
Don't forget about the changelog :)

**Review by mpneuried — APPROVED** [review]
👍

**Comment by nemanjabogdanovic** [comment]
@iStefo it's showing that merging is blocked, requiring your approval too (probably because you requested changes)

**Review by iStefo — APPROVED** [review]
👍

**Pull request merged**